### PR TITLE
Replace spatie/laravel-newsletter with drewm/mailchimp-api

### PR DESCRIPTION
**Pull request by eronisko**
laravel-newsletter has become more abstract (allowing for Spatie's Mailcoach integration). I've decided to replace it with a thinner wrapper (which laravel-newsletter uses under the hood)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist:

- [x] I have updated the [CHANGELOG](../CHANGELOG.md)
- [x] I have requested at least 1 reviewer for this PR
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by eronisko** [comment]
I have confirmed that the new integration is working, including double opt-in and GDPR consent.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 All notable changes to this project will be documented in this file[^1].
 
 ## [Unreleased]
+### Changed
+- spatie/laravel-newsletter to drewm/mailchimp-api
```
Comment (igor-kamil) [inline CHANGELOG.md:6]: why? we use that package also elsewhere... that's why I am curious.

**Review by igor-kamil — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (igor-kamil) [reply CHANGELOG.md:6]: thank you for the explanation! :)

**Pull request merged**